config/config.go

```diff
@@ -508,6 +522,49 @@ func (c *ServersetSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	return checkOverflow(c.XXX, "serverset_sd_config")
 }
 
+// KubernetesSDConfig is the configuration for Kubernetes service discovery.
+type KubernetesSDConfig struct {
+	Server           string   `yaml:"server"`
+	NodeMetricsPort  int      `yaml:"node_metrics_port,omitempty"`
+	InCluster        bool     `yaml:"in_cluster,omitempty"`
+	BearerTokenFile  string   `yaml:"bearer_token_file,omitempty"`
```

Comment:
In a Kubernetes cluster, the masters are secured in many possible ways. I've implemented the 3 most commonly used: bearer tokens (easiest to use in cluster, automatically mounted bearer token files in pods via service accounts); basic auth; & mutual TLS. I could move that to common, but I'd rather do that in a separate PR tbh as without this it's impossible for metrics to be gathered from a Kubernetes cluster - authentication is required.